types/stripe/index.d.ts

```diff
@@ -5209,6 +5209,258 @@ declare namespace Stripe {
          * card.
          */
         type ISourceCreationOptionsExtended = string | cards.ISourceCreationOptionsExtended;
+
+        type ISource = ISourceBase & (ICardSource | IOtherSource)
+        type ISourceTypes = "ach_credit_transfer" | "ach_debit" | "alipay" | "bancontat" | "card" | "card_present" | "eps" | "giropay" | "ideal" | "multibanco" | "p24" | "paper_check" | "sepa_credit_transfer" | "sepa_debit" | "sofort" | "three_d_secure";
+
+        interface IOtherSource extends ISourceBase {
+            type: ISourceTypes;
+            [k: string]: any;
```

Comment:
This is too broad. Only `ISourceType` should be allowed as keys here. And the value's type is always object like.
```ts
[source_data: ISourceType]: Record<string, any>
```
This key may need to be optional.